Google.PowerShell/Logging/GcLogCmdlets.cs

```diff
@@ -267,21 +288,8 @@ public object GetDynamicParameters()
         {
             if (_dynamicParameters == null)
             {
-                ParameterAttribute paramAttribute = new ParameterAttribute()
-                {
-                    Mandatory = false
-                };
-                ValidateSetAttribute validateSetAttribute = new ValidateSetAttribute(AllResourceTypes);
-                validateSetAttribute.IgnoreCase = true;
-                Collection<Attribute> attributes =
-                    new Collection<Attribute>(new Attribute[] { validateSetAttribute, paramAttribute });
-                // This parameter can now be thought of as:
-                // [Parameter(Mandatory = false)]
-                // [ValidateSet(validTypeValues)]
-                // public string { get; set; }
-                RuntimeDefinedParameter typeParameter = new RuntimeDefinedParameter("ResourceType", typeof(string), attributes);
                 _dynamicParameters = new RuntimeDefinedParameterDictionary();
-                _dynamicParameters.Add("ResourceType", typeParameter);
+                _dynamicParameters.Add("ResourceType", GenerateResourceTypeParameter(mandatory: false));
```

Comment:
FYI, using the name of the parameter is fine here. In the Google style guide for other languages `GenerateResourceTypeParameter(false /* mandatory */)` would also be acceptable. But this is fine if not better.